commands/bake.go

```diff
@@ -86,14 +86,18 @@ func runBake(dockerCli command.Cli, targets []string, in bakeOptions) (err error
 		}
 	}()
 
-	dis, err := getInstanceOrDefault(ctx, dockerCli, in.builder, contextPathHash)
-	if err != nil {
-		return err
-	}
-
+	var dis []build.DriverInfo
 	var files []bake.File
 	var inp *bake.Input
 
+	// instance only needed for reading remote bake files or building
+	if url != "" || !in.printOnly {
```

Comment:
I wonder if a better way to write this would be to fetch the instance "lazily"? So that we only call `getInstanceOnDefault` to get `dis` at the point of use?

Reply:
Yeah sounds good :tada:

The logic around drivers keeps getting hairier and hairier, we should definitely look into some cleanup soon :eyes:

Reply:
I have another branch to mutualize the builder/driver logic so we could lazily load an instance. Not yet ready but you can take a look: https://github.com/docker/buildx/compare/master...crazy-max:buildx:builder-pkg